File: 2025/November-11-2025/sumUniqueElements.py

# Code
# Testcase
# Test Result
# Test Result
# Leet
# 1748. Sum of Unique Elements
# Solved
# Easy
# Topics
# premium lock icon
# Companies
# Hint
# You are given an integer array nums. The unique elements of an array are the elements that appear exactly once in the array.

# Return the sum of all the unique elements of nums.
# Example 1:

# Input: nums = [1,2,3,2]
# Output: 4

from typing import List
from collections import defaultdict

nums = [1,2,3,2]

def sumUnique(nums: List[int]) -> int: 
  counts = defaultdict(int)
  sum = 0

  for i in range(len(nums)): 
    counts[nums[i]] += 1
  
  for key, val in counts.items(): 
    if val == 1: 
      sum += key
  return sum

print(sumUnique(nums))

File: 2025/September-12-2025/sumUniqueElements.js
// Easy
// Topics
// premium lock icon
// Companies
// Hint
// You are given an integer array nums. The unique elements of an array are the elements that appear exactly once in the array.

// Return the sum of all the unique elements of nums.

// Input: nums = [1,2,3,2]
// Output: 4
// Explanation: The unique elements are [1,3], and the sum is 4.

var nums = [1,2,3,2]

const sumUnique = (nums) => { 
  var map = new Map()
 
  for(let num of nums){
    map.set(num, (map.get(num) || 0 ) + 1)
  }
  var sum = 0; 

  for(let [key, val] of map){
    if(val === 1){
      sum += key
    }
  }
  return sum
}

console.log(sumUnique(nums))